partiql-spi/src/main/kotlin/org/partiql/spi/function/Function.kt

```diff
@@ -4,128 +4,59 @@ import org.partiql.spi.types.PType
-    }
-
-    /**
-     * Factory methods for standard function implementations.
-     */
-    public companion object {
-
-        /**
-         * TODO INTERNALIZE TO SPI AND REPLACE WITH A BUILDER (OR SOMETHING..)
-         *
-         * @param name
-         * @param parameters
-         * @param returns
-         * @param isNullCall
-         * @param isMissingCall
-         * @param invoke
-         * @return
-         */
-        @JvmStatic
-        public fun instance(
-            name: String,
-            parameters: Array<Parameter>,
-            returns: PType,
-            isNullCall: Boolean = true,
-            isMissingCall: Boolean = true,
-            invoke: (Array<Datum>) -> Datum,
-        ): Instance {
-            return object : Instance(
-                name,
-                Array(parameters.size) { parameters[it].getType() },
-                returns,
-                isNullCall,
-                isMissingCall,
-            ) {
-                override fun invoke(args: Array<Datum>): Datum = invoke(args)
-            }
-        }
-
-        /**
-         * TODO INTERNALIZE TO SPI AND REPLACE WITH A BUILDER (OR SOMETHING..)
-         *
-         * @param name
-         * @param parameters
-         * @param returns
-         * @param isNullCall
-         * @param isMissingCall
-         * @param invoke
-         * @return
-         */
-        @JvmStatic
-        public fun static(
-            name: String,
-            parameters: Array<Parameter>,
-            returns: PType,
-            isNullCall: Boolean = true,
-            isMissingCall: Boolean = true,
-            invoke: (Array<Datum>) -> Datum,
-        ): Function = _Function(
-            name, parameters, returns,
-            object : Instance(
-                name,
-                Array(parameters.size) { parameters[it].getType() },
-                returns,
-                isNullCall,
-                isMissingCall,
-            ) {
-                override fun invoke(args: Array<Datum>): Datum = invoke(args)
-            },
-        )
-    }
-
-    /**
-     * Private internal function implementation.
-     */
-    @Suppress("ClassName")
-    private class _Function(
-        private var name: String,
-        private var parameters: Array<Parameter>,
-        private var returns: PType,
-        private var instance: Instance,
-    ) : Function {
-        override fun getName(): String = name
-        override fun getParameters(): Array<Parameter> = parameters
-        override fun getReturnType(args: Array<PType>): PType = returns
-        override fun getInstance(args: Array<PType>): Instance = instance
-        override fun toString(): String {
-            // TODO debug strings for SqlTypeFamily
-            // val parameters = parameters.joinToString("__") { it.getType().PType.name }
-            val returnType = returns.name()
-            return "FN_${name}___${parameters}___$returnType"
-        }
-    }
+    @JvmStatic
+    fun static(
```

Comment:
nit since internal: could rename to `overload`? Not sure what `static` means here

Reply:
See https://github.com/partiql/partiql-lang-kotlin/pull/1722#discussion_r1925591763